python/pyspark/sql/connect/plan.py

```diff
@@ -1120,53 +1126,48 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
 
 
 class SQL(LogicalPlan):
-    def __init__(self, query: str, args: Optional[Union[Dict[str, Any], List]] = None) -> None:
+    def __init__(
+        self,
+        query: str,
+        args: Optional[List[Column]] = None,
```

Comment:
Irrevevant to changes proposed in this PR, I noticed a mix use of `List` and `Sequence` type hints in the file, but not necessarily indicating the mutability of variable.

Reply:
That's good to know, sounds good!